### ISSUE TITLE
Tab's not shown with -d --tabs 0

### ISSUE THREAD

**Issue by abbaswasim** [sharkdp/bat#2193]
IIUC in diff `-d` mode if I pass `--tabs 0` it should display tabs as it is but in my terminals (iTerm2 and Terminal) its converting those to spaces.

I can verify my file has tabs by using `cat` or `bat -A` as you can see in the attached image.

<img width="714" alt="Screenshot 2022-05-13 at 11 21 10" src="https://user-images.githubusercontent.com/5260991/168264213-597a3430-c92d-4822-8691-10cb059f39e7.png">
 
 I know bat is printing spaces because if I copy the content of my terminal after `bat` prints it and past it into sublime text, I can see the spaces. Attached image for reference.

<img width="867" alt="Screenshot 2022-05-13 at 11 23 46" src="https://user-images.githubusercontent.com/5260991/168264660-342876bf-61f1-4ede-a8cb-a81cacc62613.png">

**Comment by LoganDark** [comment]
Side note: it's good that you've ruled out your *terminal* converting tabs to spaces when copying, as some of them do that.

**Comment by abbaswasim** [comment]
Indeed. I have been looking through iTerm settings. There are a few things related to tabs stops but AFAICT I couldn’t find anything that talks about conversion. And either way it won’t just apply to `bat`.

**Comment by sharkdp** [comment]
It might also be the pager (`less`) which converts tabs. Can you try with `--paging=never`/`-P`?

**Comment by abbaswasim** [comment]
`--paging=never` and `-P` doesn't help.

**Comment by sharkdp** [comment]
I can not reproduce this. Works for me with `-P`.

Can you please post the output of `bat --diagnostic` in order for us to get more information about your system?

**Comment by abbaswasim** [comment]
```
#### Software version

bat 0.20.0

#### Operating system

macOS 11.6.5 (Darwin 20.6.0)

#### Command-line

```bash
bat --diagnostic
```

#### Environment variables

```bash
SHELL=/bin/zsh
PAGER=less
LESS=-R
LANG=en_GB.UTF-8
LC_ALL=<not set>
BAT_PAGER=<not set>
BAT_CACHE_PATH=<not set>
BAT_CONFIG_PATH=<not set>
BAT_OPTS=<not set>
BAT_STYLE=<not set>
BAT_TABS=0
BAT_THEME=<not set>
XDG_CONFIG_HOME=<not set>
XDG_CACHE_HOME=<not set>
COLORTERM=truecolor
NO_COLOR=<not set>
MANPAGER=<not set>
```

#### Config file

Could not read contents of '/Users/abbaswasim/.config/bat/config': No such file or directory (os error 2).

#### Compile time information

- Profile: release
- Target triple: x86_64-apple-darwin
- Family: unix
- OS: macos
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2,sse3,ssse3
- Host: x86_64-apple-darwin

#### Less version

```
> less --version
less 487 (POSIX regular expressions)
Copyright (C) 1984-2016  Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Homepage: http://www.greenwoodsoftware.com/less
```

**Comment by abbaswasim** [comment]
Is `BAT_TABS=0` of any interest? sorry I haven't customised `bat` to that extent to set environment variables etc so curious.

**Comment by abbaswasim** [comment]
By the way I have also tried it in `bash` in which case both `cat` and `bat` print spaces for tabs. So it looks like only `zsh` has the "right" behaviour for `cat`.

**Comment by sharkdp** [comment]
> `--paging=never` and `-P` doesn't help.

Okay. And does it only happen with `-d`/`--diff`?

**Comment by abbaswasim** [comment]
No just `bat` has the same behavior. I originally saw this while using `delta` And then I thought delta must be using ‘bat’s diff version so tried that. But as you said in one of your earlier comments it’s not `diff` Related

**Comment by abbaswasim** [comment]
Actually scratch that. The reason I was using diff with `tabs 0` Is because just bat on its own always prints spaces and that’s the default behavior it prints tabs only with ‘-A` AFAIK

**Comment by sharkdp** [comment]
@abbaswasim Can you please summarize the current state? is this still a problem?

**Comment by abbaswasim** [comment]
@sharkdp the status hasn't changed. The issue is still there. Its a problem for me because I can't see tab to spaces or spaces to tabs changes in my git diffs.
